src/pages/benchmarks/[benchmark_id]/index.tsx

```diff
@@ -47,27 +48,42 @@ const BenchmarkPage = () => {
     onSuccess: async () => await utils.iep.getSubgoalAndTrialData.invalidate(),
   });
 
-  const [notesValue, setNotesValue] = useState("");
+  const [notesInputValue, setNotesInputValue] = useState("");
+  const [successInputValue, setSuccessInputValue] = useState(0);
+  const [unsuccessInputValue, setUnsuccessInputValue] = useState(0);
+  const [isInputChange, setIsInputChange] = useState(false);
```

Comment:
I think this could be a computed property based on something like `const hasInputChanged = notesInputValue !== (currentTrial?.notes ?? "") || // ... other properties`?

Reply:
At first I didn't think this would work, since the initial state would be different from currentTrial. But after testing, I realized the debounce will wait until after the setState, giving it enough time to not run the unnecessary update. Good catch!